src/utils/page.js

```diff
@@ -38,6 +41,32 @@ export function getPage (win = window) {
   return detectedPageUrl
 }
 
+/**
+ * @return {string|undefined}
+ */
+export function getContextElements (contextSelectors, contextElementsLength, win = window) {
+  if (!contextSelectors || contextSelectors === '' || !contextElementsLength) {
+    return ''
+  } else {
+    var collectedElements = _collectElementsText(contextSelectors, contextElementsLength, win)
+    return base64UrlEncode(collectedElements)
+  }
+}
+
+function _collectElementsText (contextSelectors, contextElementsLength, win = window) {
+  const collectedElements = win.document.querySelectorAll(contextSelectors)
+  var i = 0
+  var collectedString = ''
+  while (i < collectedElements.length) {
+    var nextElement = replaceEmailsWithHashes(collectedElements[i].outerHTML, emailLikeNoDoubleQuotesRegex).stringWithoutRawEmails
+    var n = nextElement.length + collectedString.length
+    if (4 * Math.ceil(n / 3.0) < contextElementsLength) collectedString = collectedString + nextElement
```

Comment:
You should compare the amount of bytes in the utf-8 encoded string here, not the length of the string itself.

Reply:
where does this `4/3` proportion come from? Why does it work for bytes?

Reply:
base64 encodes 3 bytes into 4 bytes representing ASCII characters.